src/software/thunderscope/robot_diagnostics/drive_and_dribbler_widget.py

```diff
@@ -170,60 +177,59 @@ def setup_dribbler(self, title: str) -> QGroupBox:
 
         return group_box
 
-    def toggle_all(self, enable: bool) -> None:
+    def set_enabled(self, mode: ControlMode) -> None:
         """
         Disables or enables all sliders and buttons depending on boolean parameter
 
         Updates listener functions and stylesheets accordingly
         :param enable: boolean parameter, True is enable and False is disable
         """
-        if enable:
-            if not self.enabled:
-                # disconnect all sliders
-                self.disconnect_sliders()
-
-                # enable all sliders by adding listener to update label with slider value
-                common_widgets.enable_slider(
-                    self.x_velocity_slider, self.x_velocity_label, self.value_change
-                )
-                common_widgets.enable_slider(
-                    self.y_velocity_slider, self.y_velocity_label, self.value_change
-                )
-                common_widgets.enable_slider(
-                    self.angular_velocity_slider,
-                    self.angular_velocity_label,
-                    self.value_change,
-                )
-                common_widgets.enable_slider(
-                    self.dribbler_speed_rpm_slider,
-                    self.dribbler_speed_rpm_label,
-                    self.value_change,
-                )
-
-                # enable buttons
-                common_widgets.change_button_state(self.stop_and_reset_dribbler, True)
-                common_widgets.change_button_state(self.stop_and_reset_direct, True)
-
-                # change enabled field
-                self.enabled = True
-        else:
-            if self.enabled:
-                # reset slider values and disconnect
-                self.reset_all_sliders()
-                self.disconnect_sliders()
-
-                # disable all sliders by adding listener to keep slider value the same
-                common_widgets.disable_slider(self.x_velocity_slider)
-                common_widgets.disable_slider(self.y_velocity_slider)
-                common_widgets.disable_slider(self.angular_velocity_slider)
-                common_widgets.disable_slider(self.dribbler_speed_rpm_slider)
-
-                # disable buttons
-                common_widgets.change_button_state(self.stop_and_reset_dribbler, False)
-                common_widgets.change_button_state(self.stop_and_reset_direct, False)
-
-                # change enabled field
-                self.enabled = False
+        if mode == ControlMode.DIAGNOSTICS:
+            # if not self.enabled:
```

Comment:
do we want this conditional? i assume this change now means that these sliders are enabled for diagnostics and disabled for handheld mode

Reply:
I forgot to remove it completely from the class, good catch. So far in testing, the sliders are enabled and work properly in diagnostics mode, and are reset and disabled in handheld mode. Will add some screenshots soon :eyes: